Assignment 2 Finisned by Lintong Li.Rmd

```diff
@@ -0,0 +1,196 @@
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
+
+#We can change the thickness of the edge according to the number of times a particular teacher has sent a particular student to the principal.
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
+
+````
+
+## Part II
+
+In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
+
+```{r}
+library (dplyr)
+library(igraph)
+EDGE1 <- select(EDGE, from, to)
+# create a new data frame to show the relation between stid and tid.
+names(EDGE1) <- c("tid", "stid")
+S.EDGE <- count(EDGE1, stid, stid)
+S.EDGE
+names(S.EDGE) <- c("id", "count")
+T.EDGE <- count(EDGE1, tid, tid)
+T.EDGE
+names(T.EDGE) <- c("id", "count")
+VERTEX2 <- bind_rows(T.EDGE, S.EDGE)
+VERTEX2
+VERTEX <- merge(VERTEX2, VERTEX, by = "id")
+VERTEX
+g2 = graph.data.frame(EDGE, directed = TRUE, vertices = VERTEX)
+g2
+plot(g2, layout = layout.fruchterman.reingold, vertex.size = VERTEX$count*3)
+```
+```
+
+
+
+
+
+## Part III
+
+Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
+
+
+```
+```{r}
+library(igraph)
+library(dplyr)
+library(tidyr)
+# Getting the data frame from "hudk4050-classes.csv".
+D3 <- read.csv("hudk4050-classes.csv", header = TRUE)
+D3 <- as.data.frame(D3)
+D3 <- unite(D3, Name, "First.Name", "Last.Name", sep=" ")
+D3 <- filter(D3, Name != "ZIMO CHEN")
+# Getting the data frame which includes 3 colums(Name, Class.Name and count). 
+D4 <- gather(D3,Class.number, Class.Name, -Name)
+D4 <- filter(D4, Class.Name >0)
+D4 <- select(D4, Name,Class.Name)
+f1 <- function(a)
+{ gsub(" ", "", a)}
+D4 <- lapply(D4,f1)
+D4$count <- 1
+D4<- as.data.frame(D4)
+D4<-filter(D4, Class.Name!= "HUDK4050")
+#fixing the wrong information in D4. 
+D4$Class.Name <- gsub(pattern = "QMSS-",
+                                     replacement = "",
+                                     x = D4$Class.Name)
+D4$Class.Name <- gsub(pattern = "QMSS",
+                                     replacement = "G",
+                                     x = D4$Class.Name)
+D4$Class.Name <- gsub(pattern = "GG",
+                                     replacement = "G",
+                                     x = D4$Class.Name)
+D4$Class.Name <- gsub(pattern = "GR",
+                                     replacement = "G",
+                                     x = D4$Class.Name)
+# Getting the Person-Course Matrix 
```

Comment:
You can omit class HUDK 4050 from the analysis since everyone in the dataset is taking this class.